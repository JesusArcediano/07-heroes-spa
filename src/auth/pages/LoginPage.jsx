import { useContext } from "react";
import { useNavigate } from "react-router-dom"
import { AuthContext } from "../context/AuthContext";
import { PrivateRoute } from "../../router/PrivateRoute";

export const LoginPage = () => {

  const { login } = useContext( AuthContext );
  const navigate = useNavigate();

  const onLogin = () => {
    let lastPath = localStorage.getItem('lastPath') || '/';
    
    login('Fernando Herrera');

    navigate( lastPath, {
      replace: true
    });
  }


  return (
    <div className="container mt-5">
      <h1>LoginPage</h1>
      <hr />

      <button
        className="btn btn-primary"
        onClick={ onLogin }
      >
        Login
      </button>
    </div>
  )
}
