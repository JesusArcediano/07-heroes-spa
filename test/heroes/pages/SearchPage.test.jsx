import { fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter, useNavigate } from "react-router-dom";
import { SearchPage } from "../../../src/heroes/pages/SearchPage";


const mockUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockUseNavigate,
}));



describe('Pruebas en <SearchPage />', () => {


    beforeEach(()=>{ jest.clearAllMocks() });


    test('Debe mostrarse correctamente con los valores por defecto', () => {

        const { container } = render(
            <MemoryRouter>
                <SearchPage/>
            </MemoryRouter>
        );

        expect(container).toMatchSnapshot();

    });
    

    test('Debe mostrar a Batman y el input con el valor del queryString', () => {

        const { container } = render(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <SearchPage/>
            </MemoryRouter>
        );

        expect(container).toMatchSnapshot();
        const searchInput = screen.getByRole('textbox');
        expect( searchInput.value ).toBe('batman');

        const image = screen.getByRole('img');
        expect(image.src).toContain('/public/heroes/dc-batman.jpg');

    });






    test('Debe mostrar un error si no se encuentra el hero (batman123) ', () => {

        const { container } = render(
            <MemoryRouter initialEntries={['/search?q=batman123']}>
                <SearchPage/>
            </MemoryRouter>
        );

        const searchInput = screen.getByRole('textbox');
        expect( searchInput.value ).toBe('batman123');
        screen.debug();
        expect( screen.getByText('No hero with')).toBeTruthy();
        expect( screen.getByText('batman123')).toBeTruthy();
    });






    test('Debe llamar el navigate a la pantalla nueva', () => {

        const inputValue = 'superman';

        render(
            <MemoryRouter initialEntries={['/search']}>
                <SearchPage/>
            </MemoryRouter>
        );

        const input = screen.getByRole('textbox');
        fireEvent.change(
            input,
            {
                target: {
                    name: 'searchText', 
                    value: inputValue
                }
            }
        )
            
        screen.debug();
        const form = screen.getByRole('form');
        fireEvent.submit(form);

        expect( mockUseNavigate ).toHaveBeenCalledWith(`?q=${inputValue}`);

    })


})