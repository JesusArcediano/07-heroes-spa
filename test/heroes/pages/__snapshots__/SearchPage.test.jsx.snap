// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage /> Debe mostrar a Batman y el input con el valor del queryString 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value="batman"
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="col animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-4"
            >
              <img
                alt="Batman"
                class="card-img"
                src="/public/heroes/dc-batman.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-title"
              >
                Batman
              </div>
              <p
                class="card-text"
              >
                Bruce Wayne
              </p>
              <p />
              <p
                class="card-text"
              >
                <small
                  class="text-muted"
                >
                  Detective Comics #27
                </small>
              </p>
              <a
                href="/hero/dc-batman"
              >
                MÃ¡s...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pruebas en <SearchPage /> Debe mostrarse correctamente con los valores por defecto 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search a hero
      </div>
    </div>
  </div>
</div>
`;
